packages/react-search-ui/src/types/index.ts

```diff
@@ -0,0 +1,20 @@
+import { Facet, Filter } from "@elastic/search-ui";
+
+export type BaseContainerProps = {
+  children?: React.ReactNode;
+  className?: string;
+};
+
+export type BaseContainerStateProps = {
+  error: string;
+  filters: Filter[];
+  facets: { [key: string]: Facet };
+};
+
+export type Rename<
+  T,
+  R extends {
+    [K in keyof R]: K extends keyof T ? PropertyKey : "Error: key not in T";
+  }
+  // eslint-disable-next-line prettier/prettier
+> = { [P in keyof T as P extends keyof R ? R[P] : P]: T[P] }
```

Comment:
So! this TS utility aim is to rename the keys within a type. Where this will be used is for say a container component which has renamed the property when passing it to the view. See [Paging Container component](https://github.com/elastic/search-ui/blob/445dbe5a7889bb0db17f2b8b44bd0bc08d43f11e/packages/react-search-ui/src/containers/Paging.tsx#L10,L16) 
- L10 is where `setCurrent` type its being picked from the Context State
- L13-L16 is where we are using rename to create a change in the type to rename setCurrent to onChange
- l43 is where this rename is happening

The other possibilities could of been just to create a type for onChange but if setCurrent method changed, onChange type would need to be addressed too. The other way is just to rename `onChange` back to `setCurrent` but thats not feasible either.

Yep you're correct, i pulled it off SO. Il add a comment with the link :)

Reply:
added link in comments + brief description on how to use function.